api/v1alpha1/hostedcluster_types.go

```diff
@@ -217,13 +225,15 @@ type HostedClusterSpec struct {
 	// ".dockerconfigjson" whose value is the pull secret JSON.
 	//
 	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="pullSecret is immutable"
```

Comment:
Let's drop +immutable here, we want this to be mutable https://issues.redhat.com/browse/HOSTEDCP-593